lib/beetle/dead_lettering.rb

```diff
@@ -68,6 +72,58 @@ def set_queue_policy!(server, queue_name, options={})
       :ok
     end
 
+    def remove_obsolete_bindings(server, queue_name, bindings)
+      logger.debug "Removing obsolete bindings"
+      raise ArgumentError.new("server missing")     if server.blank?
+      raise ArgumentError.new("queue name missing") if queue_name.blank?
+      raise ArgumentError.new("bindings missing")   if bindings.nil?
+
+      desired_bindings = bindings.each_with_object({}) do |b, desired|
+        desired[[b[:exchange], b[:key]]] = b.except(:exchange, :key)
+      end
+
+      server_bindings = retrieve_bindings(server, queue_name)
+      server_bindings.each do |b|
+        next unless b["destination_type"] == "queue" || b["destination"] == queue_name
+        next if b["source"] == ""
+        source_route = b.values_at("source", "routing_key")
+        unless desired_bindings.has_key?(source_route)
+          logger.info "Removing obsolete binding: #{b.inspect}"
+          remove_binding(server, queue_name, b["source"], b["properties_key"])
+        end
+      end
+    end
+
+    def retrieve_bindings(server, queue_name)
+      request_url = URI("http://#{server}/api/queues/#{vhost}/#{queue_name}/bindings")
+      request = Net::HTTP::Get.new(request_url)
+
+      response = run_rabbit_http_request(request_url, request) do |http|
+        http.request(request)
+      end
+
+      unless response.code == "200"
+        log_error("Failed to retrieve bindings for queue #{queue_name}", response)
+        raise FailedRabbitRequest.new("Could not retrieve queue bindings")
+      end
+
+      JSON.parse(response.body)
+    end
+
+    def remove_binding(server, queue_name, exchange, properties_key)
+      request_url = URI("http://#{server}/api/bindings/#{vhost}/e/#{exchange}/q/#{queue_name}/#{properties_key}")
+      request = Net::HTTP::Delete.new(request_url)
+
+      response = run_rabbit_http_request(request_url, request) do |http|
+        http.request(request)
+      end
+
+      unless %w(200 201 204).include?(response.code)
+        log_error("Failed to remove obsolete binding for queue #{queue_name}", response)
+        raise FailedRabbitRequest.new("Could not retrieve queue bindings")
+      end
+    end
```

Comment:
As discussed, I think the `DeadLettering` class is now doing too much. We should find a new home for all queue management related methods.

Reply:
renamed to QueueProperties